core/trino-spi/src/main/java/io/trino/spi/predicate/TupleDomain.java

```diff
@@ -475,6 +475,30 @@ public String toString(ConnectorSession session)
         return TupleDomain.withColumnDomains(result);
     }
 
+    public <U> TupleDomain<U> transformKeys(Function<T, U> function)
+    {
+        if (isNone()) {
+            return none();
+        }
+        if (isAll()) {
+            return all();
+        }
+
+        Map<T, Domain> domains = this.domains.orElseThrow();
+        HashMap<U, Domain> result = new LinkedHashMap<>(domains.size());
+        for (Map.Entry<T, Domain> entry : domains.entrySet()) {
+            U key = function.apply(entry.getKey());
+            requireNonNull(key, () -> format("mapping function %s returned null for %s", function, entry.getKey()));
```

Comment:
does it make sense to put `function` in string? Does it ever format well?

Reply:
Depends how the function is implemented. In most cases stacktrace should be all you need. Does it hurt though?